src/preprocess/preprocess.py

```diff
@@ -17,25 +17,40 @@ def __init__(self, file):
     2. Dropping the sensitive attribute from the dataFrame
     """
 
-    def loadProcessData(self):
-        raw_input_data = pd.read_csv(self.path)
+    def loadData(self, sensitive_attribute, attribute, predictionValue, prediction_column):
+        input_data = pd.read_csv(self.path)
+        df = pd.DataFrame(input_data)
+        """ 
+            Perform the same preprocessing as the original analysis by Pro-Publica
+            https://github.com/propublica/compas-analysis/blob/master/Compas%20Analysis.ipynb
+        """
+        df = df[(df.days_b_screening_arrest <= 30)
+                & (df.days_b_screening_arrest >= -30)
+                & (df.is_recid != -1)
+                & (df.c_charge_degree != 'O')
+                & (df.score_text != 'N/A')]
+        sensitive_attribs = [sensitive_attribute]
+        Z = self.split_columns(df, sensitive_attribs, sensitive_attribute, attribute)
+        y = (df[prediction_column] == predictionValue).astype(int)
+        X = (df.drop(columns=['race', 'score_text']).fillna('Unknown').pipe(pd.get_dummies, drop_first=True))
 
-        df = pd.DataFrame(raw_input_data)
-
-        sensitive_attribs = ['race']
-        Z = (df.loc[:, sensitive_attribs].assign(race=lambda df: (df['race'] == 'African-American').astype(int)))
-
-        y = (df['score_text'] == 'High').astype(int)
+        return X, y, Z
 
-        X = (df.drop(columns=['race', 'score_text']).fillna('Unknown')
-             .pipe(pd.get_dummies, drop_first=True))
+    """
+    Split the sensitive attribute column so that this is not part of training set
+    """
+    def split_columns(self, df, sensitive_attribs, sensitive_attribute, attribute):
+        Z = (df.loc[:, sensitive_attribs].assign(
+            new_column=lambda df: (df[sensitive_attribute] == attribute).astype(int)))
+        Z.drop(columns=[sensitive_attribute], inplace=True)
+        Z.rename(columns={'new_column': sensitive_attribute}, inplace=True)
+        return Z
 
-        return X, y, Z
 
     """
     Split the data into train and test set.
     """
-    def split_data(self,X, y, Z):
+    def split_data(self, X, y, Z):
         X_train, X_test, y_train, y_test, Z_train, Z_test = train_test_split(X, y, Z, test_size=0.3, stratify=y,
                                                                              random_state=7)
```

Comment:
Is it good idea to leave `random_state` as specified in non-example code? I see that often in examples, so the reader can generate the same plots as the article, etc. Do you want to set the random state, in this context? 

I'm not 100% sure of conventions, so I'm not suggesting either way.

Reply:
yes, thank you for noticing, removed it.